python/tank_vendor/shotgun_authentication/session_cache.py

```diff
@@ -411,6 +412,12 @@ def generate_session_token(hostname, login, password, http_proxy, auth_token=Non
         raise AuthenticationError("Authentication failed.")
     except (ProtocolError, httplib2.ServerNotFoundError):
         raise AuthenticationError("Server %s was not found." % hostname)
+    except httplib2.socks.ProxyError, e:
+        logger.exception("Unexpected proxy error.")
+        raise AuthenticationError("Proxy error: %s" % str(e))
```

Comment:
Not sure the cast to str here is needed as the %s in the formatting will essentially do the same thing